back-end/routes/users.js

```diff
@@ -62,6 +62,21 @@ router.post('/login', [
   }
 });
 
+/* GET users recommendations */
+router.get('/:id/recommendations', validateSelfJWT, async function(req, res, next) {
+  // veryify permission and retrieve ID.
+  const userID = await getUserIdFromToken(req, res);
+
+  // use helper to retrieve recommendation list.
+  let responseBody = await getRecommendations(userID);
+
+  if (!responseBody) {
+    // no user data could be found
```

Comment:
I think its better just to return a HTTP 200 with an empty array. Let the frontend deal with showing no results found.

Reply:
Also haven't tested this locally yet, but doesn't rows already return an array? I don't think its necessary to return an array inside an array.

Reply:
Updated